### ISSUE TITLE
ReorderebleListView will move the Semantics focus to the incorrect position when reordering.

### ISSUE THREAD

**Issue by ghost** [flutter/flutter#100946]
## Steps to Reproduce

1. Execute `flutter run` on the code sample 
2. Turn on VoiceOver. (Only VoiceOver, No problem with Talkback)
3. Focus to Item.
4. Double tap + Long press to reorder item. 
5. Releasing the finger moves the focus to an incorrect position.

**Expected results:** Focus should be on Item.

**Actual results:** Step 5 will move the focus to the incorrect position and tapping on the Item will no longer focus it. (See attached video, the focus has shifted to the upper-left. It will be easier to understand if you turn off the mute.)

https://user-images.githubusercontent.com/8084530/160527240-4bb72706-6af9-454f-b789-8c3901ed0e07.MP4

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Reorder Test',
      theme: ThemeData(primarySwatch: Colors.blue),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key}) : super(key: key);

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  final _itemKey = const ValueKey(1);
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('Reorder Test')),
      body: ReorderableListView(
        children: [
          ListTile(
            key: _itemKey,
            title: const Text('Item'),
          ),
        ],
        onReorder: (oldIndex, newIndex) {},
      ),
    );
  }
}
```

</details>

<details>
  <summary>Logs</summary>

No particular log is displayed during operation.

flutter analyze reporting to no Issues.

```
[✓] Flutter (Channel stable, 2.10.3, on macOS 12.2.1 21D62 darwin-arm, locale ja-JP)
    • Flutter version 2.10.3 at /Users/ttada/devlib/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7e9793dee1 (4 weeks ago), 2022-03-02 11:23:12 -0600
    • Engine revision bd539267b4
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/ttada/Library/Android/sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • ANDROID_SDK_ROOT = /Users/ttada/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] VS Code (version 1.65.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.36.0

[✓] Connected device (2 available)
    • iPhoneX (mobile) • 39c14bdf121be47ecbcdd1113502f9ff63d04afe • ios            • iOS 14.6 18F72
    • Chrome (web)             • chrome                                   • web-javascript • Google Chrome 99.0.4844.83

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

**Comment by darshankawar** [comment]
@ttada 
It could be happening because there's no other item in the listview to focus on, so the focus goes back to back button by default. Also, accessibility on iOS works differently from Android.
Can you try to add more items in the list view and then follow the steps to see if the focus moves to next item on the list or goes back to the top ?

**Comment by ttada** [comment]
The same problem occurred when I implemented sorting with the number of items set to 10. The focus is not on back button, It is in the empty position above the back button. As a new discovery, the animated items seem to go wrong. For example, When I move the 4th item to the 6th item, the 4th, 5th, and 6th items will be incorrect. But when the 6th item is moved to the 4th, the 5th and 6th items will be incorrect.

https://user-images.githubusercontent.com/8084530/160595702-e7759ac4-7947-4a07-8559-c879dd61ef7d.MP4

I tried this on my iPad and it was more obvious.

https://user-images.githubusercontent.com/8084530/160597526-45e17b34-bb89-4740-9e3c-b1a3aba3a761.MP4

<details>
<summary>Reorder Code</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Reorder Test',
      theme: ThemeData(primarySwatch: Colors.blue),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key}) : super(key: key);

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  final _itemKeys = [for (var i = 0; i < 10; i++) ValueKey(i)];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
          leading: IconButton(
            icon: const Icon(Icons.arrow_back),
            onPressed: () {},
          ),
          title: const Text('Reorder Test')),
      body: ReorderableListView(
        children: [
          for (var key in _itemKeys)
            ListTile(
              key: key,
              title: Text('Item ${key.value}'),
            ),
        ],
        onReorder: (oldIndex, newIndex) {
          final item = _itemKeys.removeAt(oldIndex);
          _itemKeys.insert(newIndex - (newIndex > oldIndex ? 1 : 0), item);
          setState(() {});
        },
      ),
    );
  }
}
```
</details>

**Comment by darshankawar** [comment]
Thanks for the update. I do see the same behavior, ie, after reordering the items from the listview, the semantics focus moves to the top of the screen, beyond appbar.

<details>
<summary> stable, master flutter doctor -v </summary>

```

[✓] Flutter (Channel stable, 2.10.4, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 2.10.4 at /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c860cba910 (3 days ago), 2022-03-25 00:23:12 -0500
    • Engine revision 57d3bac3dd
    • Dart version 2.16.2
    • DevTools version 2.9.2

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[✓] Flutter (Channel master, 2.13.0-0.0.pre.296, on macOS 12.2.1 21D62
    darwin-x64, locale en-GB)
    • Flutter version 2.13.0-0.0.pre.296 at
      /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 61c30eed4f (7 minutes ago), 2022-03-28 21:25:06 -0700
    • Engine revision e21d49ae26
    • Dart version 2.17.0 (build 2.17.0-246.0.dev)
    • DevTools version 2.11.4

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>
